File: src/main/java/com/anatartari/tatamecontrolapi/infra/database/entity/SportEntity.java
package com.anatartari.tatamecontrolapi.infra.database.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.sql.Time;

@Getter
@Setter
@Entity
@Table(name = "sport")
public class SportEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String sensei;

    @Column(nullable = false)
    private Double price;

    @Column(name = "day_of_week", nullable = false)
    private String dayOfWeek;

    @Column(name = "start_time", nullable = false)
    private Time startTime;

    @Column(name = "end_time", nullable = false)
    private Time endTime;

}

File: src/sport/entities/sport.entity.ts
import { Registration } from "src/registration/entities/registration.entity";
import { Column, Entity, OneToMany } from "typeorm";
import { BaseEntity } from "../../global/entities/base.entity";

@Entity()
export class Sport extends BaseEntity {
    @Column({ length: 50, nullable: false })
    name: string;

    @Column({ length: 255, nullable: false })
    sensei: string;

    @Column({ type: 'decimal', precision: 10, scale: 2, nullable: false })
    price: number;

    @Column({ name: "day_of_week", length: 50, nullable: false })
    dayOfWeek: string;

    @Column({ name: "start_time", type: 'time', nullable: false })
    startTime: string;

    @Column({ name: "end_time", type: 'time', nullable: false })
    endTime: string;

    @Column({ name: "active", type: 'boolean', nullable: false, default: true })
    active: boolean;

    @OneToMany(() => Registration, registration => registration.sport)
    registrations: Registration[];
}
